digEmailExtractor/EE.py

```diff
@@ -0,0 +1,301 @@
+        if result.find('gmail') >= 0:
+            if result != 'gmail.com':
+                return ''
+        return result
+
+    @staticmethod
+    def clean_username(string):
+        """
+
+        :param string:
+        :type string:
+        :return:
+        :rtype:
+        """
+        username = string.strip()
+        username = re.sub("[(]?dot[)]?", '.', username)
+        # paranoid sanity check to reject short user names.
+        if len(username) < 4:
+            return None
+        return username
+
+    def clean(self, matches):
+        clean_results = Set()
+        for (u, d) in matches:
+            # print "user: %s, domain: %s" % (u, d)
+            domain = self.clean_domain(d)
+            username = EE.clean_username(u)
+            if domain and username:
+                email = username + "@" + domain
+                clean_results.add(email)
+                # print ">>> %s" % email
+        return list(clean_results)
+
+    def extract_domain(self, string):
+        """Extract the domain part of an email address within a string.
+        Separate method used for testing purposes only.
+        :param string:
+        :return:
+        :rtype:
+        """
+        matches = re.findall(self.dns_re, string, re.I)
+
+        clean_results = []
+        for m in matches:
+            clean_results.append(self.clean_domain(m))
+            # print("domains: "+', '.join(clean_results))
+        # print "\n"
+        return clean_results
+
+    def normalize(self, clean, unclean, output_format):
+        if self.output_format == EE.EE_OUTPUT_FORMAT_LIST:
+            return clean
+        else:
+            output = []
+            for co in clean:
+                email = {}
+                email['email'] = co
+                tmp_unclean = list(unclean)
+                # print co, tmp_unclean
+                if co in [username.strip() + "@" + domain.strip() for username, domain in tmp_unclean if domain and username]:
+                    # print co
+                    for tuc in tmp_unclean:
+                        username, domain = tuc
+                        tuc_string = username.strip() + "@" + domain.strip()
+                        if tuc_string == co:
+                            unclean.remove(tuc)
+                            continue
+                        domain = self.clean_domain(domain)
+                        username = EE.clean_username(username)
+                        if domain and username:
+                            email_string = username + "@" + domain
+                            if email_string == co:
+                                email['obfuscation'] = 'True'
+                    if 'obfuscation' not in email:
+                        email['obfuscation'] = 'False'
+                else:
+                    email['obfuscation'] = 'True'
+                output.append(email)
+            return output
+
+
+
+
+
+
+
+
+
+            # uc_tc = self.clean(unclean)
```

Comment:
Commented out code should be removed or captured in an if/else block if the functionality is needed.